bin/heudiconv

```diff
@@ -566,81 +605,172 @@ def convert(items, symlink=True, converter=None,
                 else:
-                    embedfunc.inputs.bids_info = None
-                embedfunc.inputs.force = True
-                embedfunc.base_dir = tmpdir
-                cwd = os.getcwd()
-                try:
-                    res = embedfunc.run()
-                    os.chmod(scaninfo, 0o0440)
-                    if with_prov:
-                        g = res.provenance.rdf()
-                        g.parse(prov_file,
-                                format='turtle')
-                        g.serialize(prov_file, format='turtle')
-                        os.chmod(prov_file, 0o0440)
-                except:
-                    os.chdir(cwd)
+                    #if not is_bids or converter != 'dcm2niix': ##uses dcm2niix's infofile
+                    embed_metadata_into_nifti(converter, is_bids, item_dicoms,
+                                              outname, outname_bids, prov_file,
+                                              scaninfo, tmpdir, with_prov)
                 os.chmod(outname, 0o0440)
 
-        if not custom_callable is None:
+
+        if custom_callable is not None:
             custom_callable(*item)
     shutil.rmtree(tmpdir)
 
 
+def tuneup_bids_json_files(json_files):
+    """Given a list of BIDS .json files, e.g. """
+    if not json_files:
+        return
+
+    # Harmonize generic .json formatting
+    for jsonfile in json_files:
+        json_ = json.load(open(jsonfile))
+        json.dump(json_, open(jsonfile, 'w'), indent=2)
+
+    # Load the beast
+    modality = basename(dirname(jsonfile))
+
+    if modality == 'fmap':
+        json_basename = '_'.join(jsonfile.split('_')[:-1])
+        # if we got by now all needed .json files -- we can fix them up
+        # unfortunately order of "items" is not guaranteed atm
+        if len(glob(json_basename + '*.json')) == 3:
+            json_phasediffname = json_basename + '_phasediff.json'
+            json_ = json.load(open(json_phasediffname))
+            # TODO: we might want to reorder them since ATM
+            # the one for shorter TE is the 2nd one!
+            # For now just save truthfully by loading magnitude files
+            lgr.info("Placing EchoTime fields into phasediff file")
+            for i in 1, 2:
+                json_['EchoTime%d' % i] = \
+                    json.load(open(json_basename + '_magnitude%d.json' % i))[
+                        'EchoTime']
+            # might have been made R/O already
+            os.chmod(json_phasediffname, 0o0660)
+            json.dump(json_, open(json_phasediffname, 'w'), indent=2)
+            os.chmod(json_phasediffname, 0o0440)
+
+        # phasediff one should contain two PhaseDiff's
+        #  -- one for original amplitude and the other already replicating what is there
+        # so let's load json files for magnitudes and
+        # place them into phasediff
+
+
+def embed_metadata_into_nifti(converter, is_bids, item_dicoms, outname,
```

Comment:
interesting... I have blindly kept this one, but then sanitized .json which it produced since it was way too heavy (removed all cs) and not neatly formatted, so I did reformatting with some really obnoxious use of `re` replacements...   here is that code:  https://github.com/dartmouth-pbs/heudiconv/blob/mvdoc-fixup-protocols/bin/heudiconv#L157